Iteration3/slave/storage.py

```diff
@@ -0,0 +1,51 @@
+import logging
+from itertools import pairwise
+
+logging.basicConfig(format="%(asctime)s - %(message)s", level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class MessageStorage:
+    messages_with_id = []
+
+    @classmethod
+    def insert_message(cls, message: str, message_id: int) -> None:
+        # Messages deduplication
+        if message_id in {item[1] for item in cls.messages_with_id}:
+            logger.info(
+                "Message has been discarded, as it's a duplicate",
+            )
+            return
+        cls.messages_with_id.insert(message_id, (message, message_id))
+        logger.info("Message with ID:%d has been saved", message_id)
+        # Sort messages by timestamp
+        cls.messages_with_id.sort(key=lambda x: x[1])
```

Comment:
Maybe should remove this line, due to the fact that we always insert them in the right order?

Reply:
Do you mean return append?